### PR TITLE
Order items on dashboard

### PR DESCRIPTION
**Pull request by danielburnley**
Add ordering to dashboard -

- Application view
  - App name & gem name
- Gem view
  - Gem name & application name
- Team view
  - Team name & application PR count (high to low)
  - When an application has the same PR count - it's ordered by name

https://trello.com/c/5ZxE3YXt/89-add-ordering-to-dashboard-views

### CODE REVIEW

File: lib/presenters/pull_requests_by_application.rb [hunk 1]
```diff
@@ -5,13 +5,14 @@ def execute(ungrouped_pull_requests)
         value.fetch(:application_name)
       end
 
-      grouped_pull_requests.map do |application_name, pull_requests|
+      pull_requests_by_application = grouped_pull_requests.map do |application_name, pull_requests|
```
Comment (adrianclay) [inline lib/presenters/pull_requests_by_application.rb]: Very minor, but not a fan of this variable name.  I'd probably get around needing to name this variable by extracting methods.  I'm sure you could come up with better names.

```
sort_by_app_name(sort_by_pr_title(grouped_pull_requests))
```

File: lib/presenters/pull_requests_by_team.rb [hunk 2]
```diff
@@ -2,24 +2,30 @@ module Presenters
   class PullRequestsByTeam
     def execute(teams:, ungrouped_pull_requests:)
       pull_requests_by_team = ungrouped_pull_requests.group_by { |pr| team_for_application(teams, pr.fetch(:application_name)) }
-      pull_requests_by_team.map do |team, pull_requests|
-        pull_requests_by_application = pull_requests.group_by { |pr| pr.fetch(:application_name) }
-
+      applications_by_team = pull_requests_by_team.map do |team, pull_requests|
```
Comment (adrianclay) [inline lib/presenters/pull_requests_by_team.rb]: Like this variable name.  This execute method feels a bit busy though.  Extracting ` pull_requests_by_team`  into a method would help I think.

File: spec/presenters/pull_requests_by_gem_spec.rb [hunk 3]
```diff
@@ -122,5 +146,49 @@
...
+          gds_api_adapters_pull_request2
+        ]
+      )
+
+      expect(result).to eq([
+        {
+          gem_name: 'gds-api-adapters',
+          pull_requests: [
+            gds_api_adapters_pull_request,
+            gds_api_adapters_pull_request2
```
Comment (adrianclay) [inline spec/presenters/pull_requests_by_gem_spec.rb]: The ordering here matches the ordering passed into `execute`.  Is that intentional?

File: spec/presenters/pull_requests_by_team_spec.rb [hunk 4]
```diff
@@ -141,6 +141,45 @@
...
+              title: 'Bump gds-api-adapters from 1.2.3 to 4.5.6',
+              url: 'https://www.github.com/alphagov/email-alert-api/pull/456',
+              opened_at: Date.parse('2018-01-01 08:00:00')
+            }
+          ]
+
+          expect(described_class.new.execute(
+                   teams: [team],
+                   ungrouped_pull_requests: pull_requests
+          )).to eq([
```
Comment (adrianclay) [inline spec/presenters/pull_requests_by_team_spec.rb]: Tests look good, you may find that asserting the full response becomes a pain if you change the structure here.  Could get around that by only inspecting the order ` application_name ` appears.

Comment (emileswarts) [reply spec/presenters/pull_requests_by_team_spec.rb]: Agreed, but would still need to dig into the structure to assert on those specific values.
Going to leave this for now but valid point.

File: spec/presenters/pull_requests_by_gem_spec.rb (hunk 3 above)
Comment (emileswarts) [reply spec/presenters/pull_requests_by_gem_spec.rb]: Good catch, updated.

**Pull request merged**